IndexedDB/idbindex_getAll.html

```diff
@@ -0,0 +1,230 @@
+  delete_request.onerror = function() {
+    assert_unreached('deleteDatabase should not fail');
+  };
+  delete_request.onsuccess = function(e) {
+    var req = indexedDB.open(dbName, dbVersion);
+    req.onsuccess = onsuccess;
+    req.onerror = function() {
+      assert_unreached('open should not fail');
+    };
+    req.onupgradeneeded = function(evt) {
+      var connection = evt.target.result;
+
+      var store = connection.createObjectStore('generated',
+            {autoIncrement: true, keyPath: 'id'});
+      var index = store.createIndex('test_idx', 'upper');
+      alphabet.forEach(function(letter) {
+        store.put({ch: letter, upper: letter.toUpperCase()});
+      });
+
+      store = connection.createObjectStore('out-of-line', null);
+      index = store.createIndex('test_idx', 'upper');
+      alphabet.forEach(function(letter) {
+        store.put({ch: letter, upper: letter.toUpperCase()}, letter);
+      });
+
+      store = connection.createObjectStore('out-of-line-not-unique', null);
+      index = store.createIndex('test_idx', 'half');
+      alphabet.forEach(function(letter) {
+        if (letter <= 'm')
+          store.put({ch: letter, half: 'first'}, letter);
+        else
+          store.put({ch: letter, half: 'second'}, letter);
+      });
+
+      store = connection.createObjectStore('out-of-line-multi', null);
+      index = store.createIndex('test_idx', 'attribs', {multiEntry: true});
+      alphabet.forEach(function(letter) {
+        attrs = [];
+        if (['a', 'e', 'i', 'o', 'u'].indexOf(letter) != -1)
+          attrs.push('vowel');
+        else
+          attrs.push('consonant');
+        if (letter == 'a')
+          attrs.push('first');
+        if (letter == 'z')
+          attrs.push('last');
+        store.put({ch: letter, attribs: attrs}, letter);
+      });
+
+      store = connection.createObjectStore('empty', null);
+      index = store.createIndex('test_idx', 'upper');
+    };
+  };
+}
+
+function createGetAllRequest(t, storeName, connection, range, maxCount) {
+    var transaction = connection.transaction(storeName, 'readonly');
+    var store = transaction.objectStore(storeName);
+    var index = store.index('test_idx');
+    // TODO(cmumford): Simplify once crbug.com/335871 is fixed.
+    var req = maxCount !== undefined ? index.getAll(range, maxCount) :
+              range !== undefined ? index.getAll(range) : index.getAll();
+    req.onerror = t.unreached_func('getAll request should succeed');
+    return req;
+}
+
+doSetup(location.pathname + '-IDBIndex.getAll', 1, function(evt) {
+    var connection = evt.target.result;
+    async_test(function(t) {
+      var req = createGetAllRequest(t, 'out-of-line', connection, 'C');
+      req.onsuccess = t.step_func(function(evt) {
+          var data = evt.target.result;
+          assert_class_string(data, 'Array', 'result should be an array');
+          assert_array_equals(data.map(e => e.ch), ['c']);
```

Comment:
These are upstreamed Blink tests where we could assume some ES6isms. It's not too uglifying to use function, so might as well.